coalib/output/ConsoleInteraction.py

```diff
@@ -641,50 +641,42 @@ def choose_action(console_printer, actions, apply_single=False):
             line = format_lines(STR_ENTER_NUMBER, symbol='[')
 
             choice = input(line)
-            if not choice:
-                return 0
             choice = str(choice)
-            if choice.isalpha():
-                choice = choice.upper()
-                choice = '(' + choice + ')'
-                if choice == '(N)':
-                    return 0
-                for i, action in enumerate(actions, 1):
-                    if choice in action.desc:
-                        return i
-            elif choice.isnumeric():
-                choice = int(choice)
-                if 0 <= choice <= len(actions):
-                    return choice
-
-            console_printer.print(format_lines(
-                'Please enter a valid letter.', symbol='['))
-
-
-def print_actions(console_printer, section, actions, failed_actions,
-                  apply_single):
-    """
-    Prints the given actions and lets the user choose.
-
-    :param console_printer: Object to print messages on the console.
-    :param actions:         A list of FunctionMetadata objects.
-    :param failed_actions:  A set of all actions that have failed. A failed
-                            action remains in the list until it is
-                            successfully executed.
-    :param apply_single:    The action that should be applied for all results.
-                            If it's not selected, has a value of False.
-    :return:                A tuple with the name member of the
-                            FunctionMetadata object chosen by the user
-                            and a Section containing at least all needed
-                            values for the action. If the user did
-                            choose to do nothing, return (None, None).
-    """
-    choice = choose_action(console_printer, actions, apply_single)
 
-    if choice == 0:
-        return None, None
+            for c in choice:
+                c = str(c)
+                actions_desc_len = len(actions_desc)
+                if c.isnumeric() or c == '0':
+                    if c == '0':
+                        actions_desc.append('Do (N)othing')
```

Comment:
Well, I need somehow to check for `Do Nothing` action. I think it was a hack already there